docs/source/user_guide_rewrite.md

```diff
@@ -0,0 +1,448 @@
+
+We will use the causal graph below to illustrate the key idea behind our method.
+Recall that we can obtain the joint distribution of variables P from their causal graph via a product of conditional distributions of each variable given its parents. To quantify the strength of an arrow from Z to Y, we define a new joint distribution PZ→Y, also called post-cutting distribution, obtained by removing the edge Z→Y, and then feeding Y with an i.i.d. copy of Z. The i.i.d. copy can be simulated, in practice, by applying random permutation to samples of Z. The strength of an arrow from Z to Y, denoted CZ→Y, is then the distance (e.g., KL divergence) between the post-cutting distribution PZ→Y and the original joint distribution P.
+CZ→Y:=DKL(P||PZ→Y)
+Note that only the causal mechanism of the target variable (PY∣X,Z in the above example) changes between the original and the post-cutting joint distribution. Therefore, any change in the marginal distribution of the target (obtained by marginalising the joint distribution) is due to the change in the causal mechanism of the target variable. This means, we can also quantify the arrow strength in terms of the change in the property of the marginal distribution (e.g. mean, variance) of the target when we remove an edge.
+
+#### Measuring arrow strengths in different units
+
+By default, arrow_strength employs KL divergence for measuring the arrow strength for categorical target, and difference in variance for continuous real-valued target. But we can also plug in our choice of measure, using the `difference_estimation_func` parameter. To measure the arrow strength in terms of the change in the mean, we could define:
+
+```
+>>> def mean_diff(Y_old, Y_new): return np.mean(Y_new) - np.mean(Y_old)
+```
+
+and then estimate the arrow strength:
+
+```
+>>> gcm.arrow_strength(causal_model, 'Y', difference_estimation_func=mean_diff)
+ {('X', 'Y'): 0.11898914602350251,
+ ('Z', 'Y'): 0.07811542095834415}
+```
+
+This is expected; in our example, the mean value of Y remains 0 regardless of whether we remove an incoming arrow. As such, the strength of incoming arrows to Y should be negligible.
+In summary, arrow strength can be measured in different units (e.g., mean, variance, bits). Therefore, we advise users to pick a meaningful unit—based on data and interpretation—to apply this method in practice.
+
+### Refuting Invertible Model (GCM only)
+
+## Learning Causal Structure
+
+Learning the causal structure, or causal graph that is, is only necessary in case we have no other means to get hold of the causal graph. If you already have the causal graph in your specific problem, you can skip this chapter and move on to [Performing Causal Tasks](#performing-causal-tasks).
+
+### IID
+
+TBD
+
+### Time Series
+
+TBD
+* * *
+
+# Performing Causal Tasks
+
+*[TODO: Every task chapter should make it clear that it relies on the causal graph constructed in* [Modeling Causal Relations](#modeling-causal-relations)*.]*
```

Comment:
agree. This is a very nice way of highlighting the graph as the common building block.